app/Models/Post.php

```diff
@@ -477,6 +477,31 @@ public function tag($tag)
             }
         }
 
+        $userId = $this->northstar_id;
+        $user = User::findOrFail($userId);
+        if ($user) {
```

Comment:
Since these now live in the same application, we can use Eloquent relationships to load this:

```suggestion
        if ($this->user) {
```

Reply:
🤯 whoa!

Reply:
when i made this update it throws a SQL error `SQLSTATE[HY000] [1040] Too many connections` maybe i missed something when writing? Seems like it's maybe pulling all users vs one?

Reply:
Huh, that's so strange! 🤔 I wonder if it's running into trouble from calling `$this->user` multiple times... what if you try holding `$user = $this->user` in memory and working with that `$user` variable?

Reply:
We're currently using a [cross-database relationship](https://github.com/jenssegers/laravel-mongodb#cross-database-relationships) to power this particular relationship, so it might not be quite as efficient as Laravel's built-in relationship support for SQL models...

Reply:
that feels right to me 👍

Reply:
so i was still running into issues with just $this->user `Undefined property: App\Observers\SignupObserver::$user` but i _think_ this is because it's in the observer rather than the model itself? So I just used `$signup->user` / `$post->user` since we are passing that in as a parameter. Seems to work as we expect.

Reply:
OH! Yeah, good catch!